### PR TITLE
TRUNK-248: Add junit tests for HibernatePatientDAO

### PR DESCRIPTION
**Pull request by vixus0**
## Description of what I changed

- Add test for getPatientIdentifiers with a list of
  PatientIdentifierTypes.
- Add test for getPatientIdentifiers with a list of Patients.
- @vixus0 @galund

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-248

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java [hunk 1]
```diff
@@ -249,16 +249,17 @@ public void deletePatientIdentifierType(PatientIdentifierType patientIdentifierT
 		// join with the patient table to prevent patient identifiers from patients
 		// that already voided getting returned
 		criteria.createAlias("patient", "patient");
+
+		// make sure the patient isn't voided
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java]: @vixus0 can you please remove this comment. the code after that says the same thing

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernatePatientDAOTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.junit.Test;
+import org.openmrs.Patient;
+import org.openmrs.PatientIdentifier;
+import org.openmrs.PatientIdentifierType;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.slf4j.Logger;
+
+public class HibernatePatientDAOTest extends BaseContextSensitiveTest {
+
+	private static final Logger log = getLogger(HibernatePatientDAOTest.class);
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernatePatientDAOTest.java]: @vixus0 you dont use the logger, can you please remove it

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java (hunk 1 above)
Comment (vixus0) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java]: done

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernatePatientDAOTest.java (hunk 2 above)
Comment (vixus0) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernatePatientDAOTest.java]: thanks, done.

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernatePatientDAOTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		List<Integer> identifierIds = identifiers.stream().map(identifier -> identifier.getId())
+				.collect(Collectors.toList());
+
+		Assert.assertEquals(2, identifiers.size());
+		Assert.assertEquals(true, identifierIds.contains(1));
+		Assert.assertEquals(true, identifierIds.contains(3));
+	}
+
+	@Test
+	public void getPatientIdentifiers_shouldGetByPatients() {
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernatePatientDAOTest.java]: @vixus0 you left the other TODO in I think, if this is the test for it

**Pull request merged**